<?xml version="1.0" ?>
<robot name="cube" xmlns:xacro="http://wiki.ros.org/xacro">
    
    <xacro:property name="cube_size" value="0.05" />
    <xacro:property name="cube_mass" value="0.1" />

    <material name="Blue">
        <color rgba="0.0 0.0 1.0 1.0" />
    </material>

    <!-- <joint name="cube_world_joint" type="floating">
        <parent link="world"/>
        <child link="cube_link"/>
        <origin rpy="0 0 0" xyz="0 -1.0 ${cube_size/2}"/> 
    </joint> -->

    <link name="cube_link">
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <geometry>
                <box size="${cube_size} ${cube_size} ${cube_size}" /> 
            </geometry>
            <material name="Blue"/>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <geometry>
                <box size="${cube_size} ${cube_size} ${cube_size}" /> 
            </geometry>
        </collision>
        <inertial>
            <mass value="${cube_mass}"/>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <inertia ixx="${(cube_mass/12.0) * (cube_size*cube_size + cube_size*cube_size)}" ixy="0" ixz="0"
                     iyy="${(cube_mass/12.0) * (cube_size*cube_size + cube_size*cube_size)}" iyz="0"
                     izz="${(cube_mass/12.0) * (cube_size*cube_size + cube_size*cube_size)}"/>
        </inertial>
    </link>
</robot>